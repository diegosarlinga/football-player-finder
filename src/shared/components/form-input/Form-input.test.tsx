import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import FormInput from './Form-input';

describe('FormInput component tests', () => {
  let wrapper: ShallowWrapper;

  describe('when rendered with no props', () => {
    beforeEach(() => {
      wrapper = shallow(<FormInput/>);
    });
  
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when rendered with props', () => {
    let onChangeSpy: jest.Mock;
    beforeEach(() => {
      onChangeSpy = jest.fn();
      const props = {
        label: 'some label',
        value: 'some value',
        name: 'some name',
        type: 'text',
        pattern: '',
        min: '12',
        max: '98',
        className: 'test-class',
        onChange: onChangeSpy,
      };
      wrapper = shallow(<FormInput {...props}/>);
    });
  
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should emit when input change', () => {
      wrapper.find('input').simulate('change');

      expect(onChangeSpy).toHaveBeenCalledTimes(1);
    });

  });
});
