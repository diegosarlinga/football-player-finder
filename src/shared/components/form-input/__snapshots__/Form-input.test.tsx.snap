// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component tests when rendered with no props should match snapshot 1`] = `
<div
  className="Form-input-container form-group w-100 undefined"
>
  <label
    className="w-100"
  >
    <input
      className="form-control Form-input"
    />
  </label>
</div>
`;

exports[`FormInput component tests when rendered with props should match snapshot 1`] = `
<div
  className="Form-input-container form-group w-100 test-class"
>
  <label
    className="w-100"
  >
    some label
    <input
      className="form-control Form-input"
      max="98"
      min="12"
      name="some name"
      onChange={[MockFunction]}
      pattern=""
      placeholder="some label"
      type="text"
      value="some value"
    />
  </label>
</div>
`;
